src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -95,6 +99,28 @@ CHIP_ERROR GeneralCommissioningAttrAccess::Read(const ConcreteReadAttributePath
     case SupportsConcurrentConnection::Id: {
         return ReadSupportsConcurrentConnection(aEncoder);
     }
+#if CHIP_CONFIG_TC_REQUIRED
+    case TCAcceptedVersion::Id: {
+        Optional<TermsAndConditions> outTermsAndConditions;
+        Server::GetInstance().GetTermsAndConditionsProvider()->GetAcceptance(outTermsAndConditions);
+        return !outTermsAndConditions.HasValue() ? aEncoder.Encode(0) : aEncoder.Encode(outTermsAndConditions.Value().version);
+    }
+    case TCMinRequiredVersion::Id: {
+        Optional<TermsAndConditions> outTermsAndConditions;
+        Server::GetInstance().GetTermsAndConditionsProvider()->GetRequirements(outTermsAndConditions);
+        return !outTermsAndConditions.HasValue() ? aEncoder.Encode(0) : aEncoder.Encode(outTermsAndConditions.Value().version);
+    }
+    case TCAcknowledgements::Id: {
+        Optional<TermsAndConditions> outTermsAndConditions;
+        Server::GetInstance().GetTermsAndConditionsProvider()->GetAcceptance(outTermsAndConditions);
+        return !outTermsAndConditions.HasValue() ? aEncoder.Encode(0) : aEncoder.Encode(outTermsAndConditions.Value().value);
+    }
+    case TCAcknowledgementsRequired::Id: {
+        Optional<TermsAndConditions> outTermsAndConditions;
+        Server::GetInstance().GetTermsAndConditionsProvider()->GetRequirements(outTermsAndConditions);
+        return aEncoder.Encode(outTermsAndConditions.HasValue());
```

Comment:
In this code, it seems that the TCAcknowledgementsRequired attribute indicates whether the return value of GetRequirements has valid value (there are requirements of terms and conditions acknowledgements).
However, in spec, it is said that this attribute indicates whether SetTCAcknowledgements is currently required to be called (it is set to True when the user doesn't accept the current version of terms and conditions).

Reply:
Yeah, I believe you are correct, I'll update this. We will update attribute read to indicate if _SetTCAcknowledgements_ is required before first _CommissiongComplete_ fill succeed or not. For secondary commissioning, we will not require any T&C check before _CommissiongComplete_, so it will always return OK.